wallet/src/main/scala/org/bitcoins/wallet/api/HDWalletApi.scala

```diff
@@ -0,0 +1,519 @@
+package org.bitcoins.wallet.api
+
+import org.bitcoins.commons.jsonmodels.wallet.CoinSelectionAlgo
+import org.bitcoins.core.currency.CurrencyUnit
+import org.bitcoins.core.hd.{AddressType, HDAccount, HDChainType, HDPurpose}
+import org.bitcoins.core.protocol.transaction.{
+  Transaction,
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.core.protocol.{BitcoinAddress, BlockStamp}
+import org.bitcoins.core.wallet.fee.FeeUnit
+import org.bitcoins.core.wallet.utxo.{AddressTag, TxoState}
+import org.bitcoins.keymanager.KeyManagerParams
+import org.bitcoins.wallet.Wallet
+import org.bitcoins.wallet.models.{AccountDb, AddressDb, SpendingInfoDb}
+
+import scala.concurrent.Future
+
+/**
+  * API for the wallet project.
+  *
+  * This wallet API is BIP44 compliant.
+  *
+  * @see [[https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki BIP44]]
+  */
+trait HDWalletApi extends WalletApi {
```

Comment:
Is there any way we can minimize the number of overloaded methods in here? There is like 4 different versions of `sendFromOutPoints`, `sendWithAlgo` etc

Reply:
Would like to but we can't set default arguments because the function is overloaded.